scan/context.go

```diff
@@ -121,26 +123,39 @@ func (s *Scanner) getContextFromReader(r io.Reader) (err error) {
 }
 
 func (s *Scanner) getContextFromRegistry(ctx context.Context, registryArtifact string) (err error) {
-	authorizer := docker.NewAuthorizer(http.DefaultClient, func(hostName string) (string, string, error) {
-		// If no matching credential found, attempt an anonymous pull
-		if s.credentials[hostName] == nil {
-			return "", "", nil
-		}
-		return s.credentials[hostName].Username.ResolvedValue, s.credentials[hostName].Password.ResolvedValue, nil
-	})
-	resolver := docker.NewResolver(docker.ResolverOptions{
-		PlainHTTP: false,
-		Hosts: docker.ConfigureDefaultRegistries(
-			docker.WithAuthorizer(authorizer),
-		),
-	})
-	fileStore := content.NewFileStore(s.destinationFolder)
-	defer fileStore.Close()
+	src, err := remote.NewRepository(registryArtifact)
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse artifact %s", registryArtifact)
+	}
+	src.Client = &auth.Client{
+		Header: http.Header{
+			"User-Agent":           {"oras-go"},
+			"X-Meta-Source-Client": {"azure/acr/tasks"},
+		},
+		Cache: auth.DefaultCache,
+		Credential: func(ctx context.Context, registry string) (auth.Credential, error) {
+			// If no matching credential found, attempt an anonymous pull
+			if s.credentials[registry] == nil {
+				return auth.EmptyCredential, nil
+			}
+
+			return auth.Credential{
+				Username: s.credentials[registry].Username.ResolvedValue,
```

Comment:
Looks like I forgot to add a nil check for the `Username` and `Password` field when I wrote the code a originally. Do you think we need to add it? I guess maybe it hasn’t been a problem so far?

Reply:
It doesn't look it can be nil

https://github.com/Azure/acr-builder/blob/d448eceff8c1117524f6d8bcfdec01ad54d05e68/graph/task.go#L441

Reply:
Ahh ok nevermind then.